.github/workflows/theme-check.yml

```diff
@@ -0,0 +1,236 @@
+name: WordPress Theme Check
+
+on:
+  pull_request:
+    types: [opened, labeled, synchronize]
+
+# Add this permissions block
+permissions:
+  pull-requests: write
+
+jobs:
+  theme-check:
+    runs-on: ubuntu-latest
+    
+    services:
+      wordpress:
+        image: wordpress:php8.2
+        ports:
+          - 8080:80
+        env:
+          WORDPRESS_DB_HOST: db
+          WORDPRESS_DB_USER: wordpress
+          WORDPRESS_DB_PASSWORD: wordpress
+          WORDPRESS_DB_NAME: wordpress
+      db:
+        image: mysql:8.0
+        env:
+          MYSQL_ROOT_PASSWORD: rootpassword
+          MYSQL_DATABASE: wordpress
+          MYSQL_USER: wordpress
+          MYSQL_PASSWORD: wordpress
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Install WP-CLI inside WordPress container
+      run: |
+        docker exec $(docker ps -qf "ancestor=wordpress:php8.2") curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+        docker exec $(docker ps -qf "ancestor=wordpress:php8.2") chmod +x wp-cli.phar
+        docker exec $(docker ps -qf "ancestor=wordpress:php8.2") mv wp-cli.phar /usr/local/bin/wp
```

Comment:
Could we use `@wordpress/env` instead? I tested the theme-check cli integration using `npx @wordpress/env run cli`